cmd/fluxctl/release_cmd.go

```diff
@@ -60,16 +60,23 @@ func (opts *serviceReleaseOpts) RunE(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	if err := checkExactlyOne("--service=<service>, or --all", opts.service != "", opts.allServices); err != nil {
+	if err := checkExactlyOne("--service=<service>, or --all", len(opts.services) > 0, opts.allServices); err != nil {
 		return err
 	}
 
-	service, err := parseServiceOption(opts.service) // will be "" iff opts.allServices
-	if err != nil {
-		return err
+	var services []flux.ServiceSpec
+	if opts.allServices {
+		services = []flux.ServiceSpec{flux.ServiceSpecAll}
+	} else {
+		for _, spec := range opts.services {
+			services = append(services, flux.ServiceSpec(spec))
```

Comment:
Would it prevent a few mishaps if this parsed the spec as a `ServiceID` before turning it into a `ServiceSpec`? Not a big deal, since this will happen at the server anyway.